Fixed issues in Drizzle and Contract Events guide (#1113)
trufflesuite/trufflesuite.com: src/guides/drizzle-and-contract-events/index.md

Now that we have a test chain running and our smart contracts deployed, let's add a toast notification to the UI.

## Listen for Contract Events

We want to listen for the `SimpleStorage` contract's `StorageSet` event and show a toast notification once it fires.

The front end code is located under the `app` folder. Lets add the notification library `react-toastify` to simulate an external interaction From the project directory:

```shell
cd app
npm install react-toastify
```

For the sake of simplicity, we will work in one file, `./app/src/middleware/index.js`.

```shell
mkdir ./src/middleware
touch ./src/middleware/index.js
```

In `./app/src/middleware/index.js`, import `EventActions` and `generateStore` from Drizzle as well as
toast from `react-toastify`, and `drizzleOptions`.

```javascript
// ./app/src/middleware/index.js
import { generateStore, EventActions } from '@drizzle/store'
import drizzleOptions from '../drizzleOptions'
import { toast } from 'react-toastify'
```

> The action `EventActions.EVENT_FIRED` is emitted whenever a contract event is detected in a Block. We will gain access to it by registering a custom middleware with the Redux store. As you may know, Redux middleware comprises a set of functions, executed in a sequence, that process each dispatched action before passing it to Reducers. Add the following:

Reviewer: I feel like Block doesn't need to be capitalized? But if it also capitalized elsewhere please ignore this!

```suggestion
The action `EventActions.EVENT_FIRED` is emitted whenever a contract event is detected in a block. We will gain access to it by registering a custom middleware with the Redux store. As you may know, Redux middleware comprises a set of functions, executed in a sequence, that process each dispatched action before passing it to Reducers. Add the following:
```

$ touch ./src/middleware/index.js
```

> In `./app/src/middleware/index.js` import `EventActions` and `generateStore` from Drizzle as well as

Reviewer: ```suggestion
In `./app/src/middleware/index.js`, import `EventActions` and `generateStore` from Drizzle as well as
```

})
```

Your completed `./app/src/middleware/index.js` file should look something like this:

```javascript
import { generateStore, EventActions } from '@drizzle/store'
import drizzleOptions from '../drizzleOptions'
import { toast } from 'react-toastify'

const contractEventNotifier = store => next => action => {
  if (action.type === EventActions.EVENT_FIRED) {
    const contract = action.name
    const contractEvent = action.event.event
    const message = action.event.returnValues._message
    const display = `${contract}(${contractEvent}): ${message}`

    toast.success(display, { position: toast.POSITION.TOP_RIGHT })
  }
  return next(action)
}

const appMiddlewares = [ contractEventNotifier ]

export default generateStore({
  drizzleOptions,
  appMiddlewares,
  disableReduxDevTools: false  // enable ReduxDevTools!
})
```

## Connect the Store

The first thing we need to do is send the `store` as a prop to `DrizzleProvider`. Add the following to `./app/src/App.js`:

```javascript
// ./app/src/App.js
...
import store from './middleware'
...
const drizzle = new Drizzle(drizzleOptions, store);
...
```

> You completed `App.js` file should looks something like this:

Reviewer: ```suggestion
Your completed `App.js` file should looks something like this:
```

  return (
    <div className="App">
      <ToastContainer />
      <div>
        <img src={logo} alt="drizzle-logo" />
        <h1>Drizzle Examples</h1>
        <p>
          Examples of how to get started with Drizzle in various situations.
        </p>
      </div>

      <div className="section">
        <h2>Active Account</h2>
        <AccountData
          drizzle={drizzle}
          drizzleState={drizzleState}
          accountIndex={0}
          units="ether"
          precision={3}
        />
      </div>

      <div className="section">
        <h2>SimpleStorage</h2>
        <p>
          This shows a simple ContractData component with no arguments, along
          with a form to set its value.
        </p>
        <p>
          <strong>Stored Value: </strong>
          <ContractData
            drizzle={drizzle}
            drizzleState={drizzleState}
            contract="SimpleStorage"
            method="storedData"
          />
        </p>
        <ContractForm drizzle={drizzle} contract="SimpleStorage" method="set" />
      </div>

      ...

    </div>
  );
};

```

## A Quick Test
  * Things often go south during development so a pretest check is in order.

>     1. MetaMask should NOT be on Main net! Do not run this if you're on main

Reviewer: ```suggestion
    1. MetaMask should NOT be on Mainnet! Do not run this if you're on Mainnet!
```

>        net!

Reviewer: ```suggestion
```
